File: wms-fe/src/services/locationService.js
import axiosClient from '../api/axiosClient';

const locationService = {
  getAll: (params) => {
    return axiosClient.get('/locations', { params });
  },
  
  getById: (id) => {
    return axiosClient.get(`/locations/${id}`);
  },
  
  create: (data) => {
    return axiosClient.post('/locations', data);
  },
  
  update: (id, data) => {
    return axiosClient.put(`/locations/${id}`, data);
  },
  
  delete: (id) => {
    return axiosClient.delete(`/locations/${id}`);
  }
};

export default locationService;

File: wms-be/src/main/java/com/restapi/wmsservice/service/LocationService.java
package com.restapi.wmsservice.service;

import com.restapi.wmsservice.dto.request.LocationRequest;
import com.restapi.wmsservice.dto.response.LocationResponse;

import java.util.List;

public interface LocationService {
    LocationResponse create(LocationRequest request);
    List<LocationResponse> getAll();
    List<LocationResponse> getByWarehouseId(Long warehouseId);
    LocationResponse getById(Long id);
    LocationResponse update(Long id, LocationRequest request);
    void delete(Long id);
}
